backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadService.java

```diff
@@ -114,48 +120,69 @@ public PatientBulkUploadResponse processPersonCSV(InputStream csvStream, UUID fa
-            extractedData.getLastName().getValue(),
-            extractedData.getSuffix().getValue(),
-            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
-            address,
-            country,
-            List.of(
-                new PhoneNumber(
-                    parsePhoneType(extractedData.getPhoneNumberType().getValue()),
-                    extractedData.getPhoneNumber().getValue())),
-            parsePersonRole(extractedData.getRole().getValue(), false),
-            List.of(extractedData.getEmail().getValue()),
-            convertRaceToDatabaseValue(extractedData.getRace().getValue()),
-            convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
-            null,
-            convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
-            parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
-            parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
-            null,
-            null);
+        // create new person with current organization, then add to new patients list
+        Person newPatient =
+            new Person(
+                currentOrganization,
+                assignedFacility,
+                null,
+                extractedData.getFirstName().getValue(),
+                extractedData.getMiddleName().getValue(),
+                extractedData.getLastName().getValue(),
+                extractedData.getSuffix().getValue(),
+                parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+                address,
+                country,
+                parsePersonRole(extractedData.getRole().getValue(), false),
+                List.of(extractedData.getEmail().getValue()),
+                convertRaceToDatabaseValue(extractedData.getRace().getValue()),
+                convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
+                null,
+                convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
+                parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
+                parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
+                null,
+                null);
+
+        // collect phone numbers and associate them with the patient, then add to phone numbers list
+        List<PhoneNumber> newPhoneNumbers =
+            _personService.assignPhoneNumbersToPatient(
+                newPatient,
+                List.of(
+                    new PhoneNumber(
+                        parsePhoneType(extractedData.getPhoneNumberType().getValue()),
+                        extractedData.getPhoneNumber().getValue())));
+        newPhoneNumbers.forEach(phoneNumber -> phoneNumbersList.add((phoneNumber)));
+
+        // set primary phone number
+        if (!newPhoneNumbers.isEmpty()) {
+          newPatient.setPrimaryPhone(newPhoneNumbers.get(0));
+        }
```

Comment:
This can also be re-written in functional style:
`newPhoneNumbers.stream().findFirst().ifPresent(newPatient::setPrimaryPhone);`

Reply:
✅